x/validator-preference/README.md

```diff
@@ -0,0 +1,155 @@
+# validator-set Preference 
+
+## Abstract 
+
+Validator-Set preference is a new module which gives users and contracts a 
+better UX for staking to a set of validators. For example: a one click button
+that stakes to multiple validators. Then the user can set (or realistically a frontend provides) 
+a list of recommended defaults (Ex: governors, wosmongton, chain/stack contributors etc).
+Currently this can be done on-chain with frontends, but having a preference list stored locally 
+eases frontend code burden. 
+
+## Design 
+
+How does this module work? 
+
+- Allow a user to set a list of {val-addr, weight} in the state, called their validator-set preference.
+- Allow a user to update a list of {val-addr, weight} in the state, then do the following; 
+  - Unstake the existing tokens (run the same unbond logic as cosmos-sdk staking).
+  - Update the validator distribution weights.
+  - Stake the tokens based on the new weights.
+- Give users a single message to stake {X} tokens, according to their validator-set preference distribution.
+- Give users a single message to unstake {X} tokens, according to their validator-set preference distribution.
+- Give users a single message to claim rewards from everyone on their preference list.
+- If a user does not have a validator-set preference list set and has is attempting to stake do the following; 
+  - Make their preference list default to their current staking distribution.
+- If a user has no preference list and no staking, then return error for messages.
+
+## Calculations
+
+Staking Calculation 
+
+- The user provides an amount to stake and our `MsgStakeToValidatorSet` divides the amount based on validator weight distribution.
+  For example: Stake 100osmo with validator-set {ValA -> 0.5, ValB -> 0.3, ValC -> 0.2}
+  our stake logic will attempt to stake (100 * 0.5) 50osmo for ValA , (100 * 0.3) 30osmo from ValB and (100 * 0.2) 20osmo from ValC.
+
+UnStaking Calculation 
```

Comment:
Lets update the staking / unstaking terminology to either:
* delegating / undelegating
or
* bonding / unbonding